model/src/main/kotlin/ResolvedConfiguration.kt

```diff
@@ -29,9 +32,47 @@ import com.fasterxml.jackson.annotation.JsonInclude
  */
 data class ResolvedConfiguration(
     /**
-     * All package curations applicable to the packages contained in the enclosing [OrtResult] ordered
-     * highest-priority-first.
+     * The curations for all enabled providers ordered highest-priority-first. The list contains exactly one entry for
+     * each enabled provider.
      */
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
-    val packageCurations: List<PackageCuration> = emptyList()
-)
+    val packageCurations: List<PackageCurationsEntry> = emptyList(),
+) {
+    init {
+        val duplicateProviderIds = packageCurations.getDuplicates().map { it.provider.id }
+        require(duplicateProviderIds.isEmpty()) {
+            "The list 'providers' contains the following duplicates, which is not allowed: " +
+                    "${duplicateProviderIds.joinToStringSingleQuoted()}."
+        }
+    }
+
+    /**
+     * Return all [PackageCuration]s contained in this [ResolvedConfiguration] in highest-priority-first order.
+     */
+    @JsonIgnore
+    fun getAllPackageCurations(): List<PackageCuration> = packageCurations.flatMap { it.curations }
+}
+
+data class PackageCurationsEntry(
```

Comment:
I would propose `ResolvedPackageCurations` as name for the class, it matches the context where it is used.

Reply:
@mnonnenmacher  are you ok if I make this a commit on top ?

Reply:
Yes